sdk/clientcore/http-stress/src/main/java/io/clientcore/http/stress/HttpGet.java

```diff
@@ -65,13 +66,82 @@ private void runInternal() {
         try (Response<?> response = pipeline.send(request)) {
             response.getBody().toBytes();
         } catch (IOException e) {
-            new UncheckedIOException(e);
+            LOGGER.logThrowableAsError(new UncheckedIOException(e));
         }
     }
 
     @Override
     public Mono<Void> runAsync() {
-        return Mono.error(new UnsupportedOperationException("Not implemented"));
+        return TELEMETRY_HELPER.instrumentRunAsync(runInternalAsync());
+    }
+
+    @Override
+    public CompletableFuture<Void> runAsyncWithCompletableFuture() {
+        return TELEMETRY_HELPER.instrumentRunAsyncWithCompletableFuture(runAsyncWithCompletableFutureInternal());
+    }
+
+    @Override
+    public Runnable runAsyncWithExecutorService() {
+        return TELEMETRY_HELPER.instrumentRunAsyncWithRunnable(runAsyncWithExecutorServiceInternal());
+    }
+
+    @Override
+    public Runnable runAsyncWithVirtualThread() {
+        return TELEMETRY_HELPER.instrumentRunAsyncWithRunnable(runAsyncWithVirtualThreadInternal());
+    }
+
+    private Mono<Void> runInternalAsync() {
+        return Mono.usingWhen(Mono.fromCallable(() -> pipeline.send(createRequest())),
+            response -> {
+                ((HttpResponse<?>) response).getBody().toBytes();
+                return Mono.empty();
+            },
+            response -> Mono.fromRunnable(() -> {
+                try {
+                    response.close();
+                } catch (IOException e) {
+                    LOGGER.logThrowableAsError(new UncheckedIOException(e));
+                }
+            }));
+    }
+
+    // Method to run using CompletableFuture
+    private CompletableFuture<Void> runAsyncWithCompletableFutureInternal() {
+        return CompletableFuture.supplyAsync(() -> {
```

Comment:
May want to change the testing design slightly where the ExecutorService overload of `supplyAsync` is used (`CompletableFuture.supplyAsync(Supplier<U>, ExecutorService)`). By default, `CompletableFuture.supplyAsync(Supplier<U>)` will use the `ForkJoinPool.common()` if there is more than one thread in the environment or, worse (for performance but better for deadlocking issues), if there is only a single thread this uses an `ExecutorService` implementation that creates a new thread per invocation.

Using a specific `ExecutorService` will get us away from contention in the common `ForkJoinPool` and prevent us from going down a path where we spawn a thread per operation which is extremely costly.